src/PowerShellEditorServices/Extensions/EngineExtensions/EditorEngine.cs

```diff
@@ -0,0 +1,117 @@
+
+        private static readonly Lazy<object> s_psesAsmLoadContextLazy = new Lazy<object>(GetPsesAsmLoadContext);
+
+        private static readonly Lazy<Type> s_asmLoadContextType = new Lazy<Type>(() => Type.GetType("System.Runtime.Loader.AssemblyLoadContext"));
+
+        private static readonly Lazy<Func<IDisposable>> s_enterPsesReflectionContextLazy = new Lazy<Func<IDisposable>>(GetPsesAlcReflectionContextEntryFunc);
+
+        private static readonly Lazy<Func<string, Assembly>> s_loadAssemblyInPsesAlc = new Lazy<Func<string, Assembly>>(GetPsesAlcLoadAsmFunc);
+
+        private static Type AsmLoadContextType => s_asmLoadContextType.Value;
+
+        private static object PsesAssemblyLoadContext => s_psesAsmLoadContextLazy.Value;
+
+        private static Func<IDisposable> EnterPsesAlcReflectionContext => s_enterPsesReflectionContextLazy.Value;
+
+        private static Func<string, Assembly> LoadAssemblyInPsesAlc => s_loadAssemblyInPsesAlc.Value;
+
+        private readonly IServiceProvider _serviceProvider;
+
+        internal EditorEngine(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+            LanguageServer = new EditorLanguageServer(_serviceProvider.GetService<ILanguageServer>());
+        }
+
+        public EditorLanguageServer LanguageServer { get; }
+
+        public object GetService(string psesServiceFullTypeName) => GetService(psesServiceFullTypeName, "Microsoft.PowerShell.EditorServices");
+
+        public object GetService(string fullTypeName, string assemblyName)
+        {
+            string asmQualifiedName = $"{fullTypeName}, {assemblyName}";
+            return GetServiceByAssemblyQualifiedName(asmQualifiedName);
+        }
+
+        public object GetService(Type serviceType)
+        {
+            return _serviceProvider.GetService(serviceType);
+        }
+
+        public object GetServiceByAssemblyQualifiedName(string asmQualifiedTypeName)
+        {
+            Type serviceType;
+            if (VersionUtils.IsNetCore)
+            {
+                using (EnterPsesAlcReflectionContext())
+                {
+                    serviceType = s_psesAsm.GetType(asmQualifiedTypeName);
+                }
+            }
+            else
+            {
+                serviceType = Type.GetType(asmQualifiedTypeName);
+            }
+
+            return _serviceProvider.GetService(serviceType);
+        }
+
+        public object GetPsesAssemblyLoadContext()
+        {
+            return PsesAssemblyLoadContext;
+        }
+
+        public Assembly LoadAssemblyInPsesLoadContext(string assemblyPath)
+        {
+            if (!VersionUtils.IsNetCore)
+            {
+                return Assembly.LoadFrom(assemblyPath);
+            }
+
+            return LoadAssemblyInPsesAlc(assemblyPath);
+        }
+
+        private static Func<IDisposable> GetPsesAlcReflectionContextEntryFunc()
+        {
+            PsesAssemblyLoadContext.GetType();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the return value of method 'GetType'.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4979332)